Chat App/View/Pub/PubRegisterViewController.swift

```diff
@@ -0,0 +1,384 @@
+            .map { [unowned self] in
+                self.viewModel.inputs.verifyUserInput(userInput: $0!)
+            }
+        
+        usernameValid
+            .skip(1)
+            .subscribe(onNext: { [weak self] in
+                guard let self = `self` else { return }
+                if $0 {
+                    self.viewModel.inputs.verifyUsernameAvailability(userInput: self.usernameField.text!)
+                } else {
+                    self.usernameMessageLabel.textColor = .red
+                    self.usernameMessageLabel.text = Constants.PubStrings.Warnings.usernameInvalidMessage
+                }
+            })
+            .disposed(by: disposeBag)
+
+        let passwordValid = passwordField
+            .rx
+            .text
+            .observe(on: MainScheduler.asyncInstance)
+            .distinctUntilChanged()
+            .throttle(.milliseconds(inputThrottleInMilliseconds), scheduler: MainScheduler.instance)
+            .map { [unowned self] in
+                self.viewModel.inputs.verifyUserInput(userInput: $0!)
+            }
+        
+        passwordValid
+            .skip(1)
+            .subscribe(onNext: { [weak self] in
+                guard let self = `self` else { return }
+                if $0 {
+                    self.passwordMessageLabel.textColor = .blue
+                    self.passwordMessageLabel.text = Constants.PubStrings.passwordValidMessage
+                    self.viewModel.passwordInput = self.passwordField.text!
+                } else {
+                    self.passwordMessageLabel.textColor = .red
+                    self.passwordMessageLabel.text = Constants.PubStrings.Warnings.passwordInvalidMessage
+                }
+            })
+            .disposed(by: disposeBag)
+
+        let confirmPasswordValid = confirmPasswordField
+            .rx
+            .text
+            .observe(on: MainScheduler.asyncInstance)
+            .throttle(.milliseconds(inputThrottleInMilliseconds), scheduler: MainScheduler.instance)
+            .map { [unowned self] in
+                self.viewModel.inputs.verifyPasswordMatch(userInput: $0!)
+            }
+        
+        confirmPasswordValid
+            .skip(1)
+            .subscribe(onNext: { [weak self] in
+                guard let self = `self` else { return }
+                if $0 {
+                    self.confirmPasswordMessageLabel.textColor = .blue
+                    self.confirmPasswordMessageLabel.text = Constants.PubStrings.passwordMatchMessage
+                } else {
+                    self.confirmPasswordMessageLabel.textColor = .red
+                    self.confirmPasswordMessageLabel.text = Constants.PubStrings.Warnings.passwordMismatchMessage
+                }
+            })
+            .disposed(by: disposeBag)
+        
+        let everythingValid = Observable
+            .combineLatest(usernameValid, passwordValid, confirmPasswordValid) {
```

Comment:
map? `).map { $0 && $1 && $2 }`

Reply:
This issue has been addressed in the most recent commits.